### PR TITLE
Bug 1916897: add test to verify build /run fs contents 

### PR DESCRIPTION
**Pull request by gabemontero**

### CODE REVIEW

**Comment by gabemontero** [comment]
GCP IAM pain

/test e2e-gcp-builds

**Comment by gabemontero** [comment]
Sigh ... forgot about the timestamps the build log adds:

```
    2021-01-20T17:10:47.581005957Z /run:
    2021-01-20T17:10:47.581005957Z lock
    2021-01-20T17:10:47.581005957Z rhsm
    2021-01-20T17:10:47.581005957Z secrets
    2021-01-20T17:10:47.581005957Z 
    2021-01-20T17:10:47.581005957Z /run/lock:
    2021-01-20T17:10:47.581005957Z 
    2021-01-20T17:10:47.581005957Z /run/rhsm:
    2021-01-20T17:10:47.581005957Z 
    2021-01-20T17:10:47.581005957Z /run/secrets:
    2021-01-20T17:10:47.581005957Z kubernetes.io
    2021-01-20T17:10:47.581005957Z openshift.io
    2021-01-20T17:10:47.581005957Z rhsm
    2021-01-20T17:10:47.581005957Z 
    2021-01-20T17:10:47.581005957Z /run/secrets/kubernetes.io:
    2021-01-20T17:10:47.581005957Z serviceaccount
    2021-01-20T17:10:47.581005957Z 
    2021-01-20T17:10:47.581005957Z /run/secrets/kubernetes.io/serviceaccount:
    2021-01-20T17:10:47.581005957Z ca.crt
    2021-01-20T17:10:47.581005957Z namespace
    2021-01-20T17:10:47.581005957Z service-ca.crt
    2021-01-20T17:10:47.581005957Z token
    2021-01-20T17:10:47.581005957Z 
    2021-01-20T17:10:47.581005957Z /run/secrets/openshift.io:
    2021-01-20T17:10:47.581005957Z pull
    2021-01-20T17:10:47.581005957Z 
    2021-01-20T17:10:47.581005957Z /run/secrets/openshift.io/pull:
    2021-01-20T17:10:47.581005957Z 
    2021-01-20T17:10:47.581005957Z /run/secrets/rhsm:
    2021-01-20T17:10:47.581005957Z ca
    2021-01-20T17:10:47.581005957Z 
    2021-01-20T17:10:47.581005957Z /run/secrets/rhsm/ca:
    2021-01-20T17:10:47.581005957Z redhat-entitlement-authority.pem
    2021-01-20T17:10:47.581005957Z redhat-uep.pem
```

Either need to see if I can turn that off, or switch to string suffix checks

**Comment by gabemontero** [comment]
OK we have a clean run of e2e-gcp-builds now at https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/25810/pull-ci-openshift-origin-master-e2e-gcp-builds/1351981457559523328 with my attempt at adding the "what is mounted in /run verification test" that @smarterclayton asked for .... it is in the passing list 

let the dissection begin

/assign @bparees 

/retest

File: test/extended/builds/run_fs_verification.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+rhsm
+secrets
+
+/run/lock:
+
+/run/rhsm:
+
+/run/secrets:
+kubernetes.io
+openshift.io
```
Comment (bparees) [inline test/extended/builds/run_fs_verification.go:46]: ultimately the goal is that these 2 dirs should not show up.  So i think the test is fine, but let's hold it until we can land it w/ the correct set of expected dirs.

Comment (gabemontero) [reply test/extended/builds/run_fs_verification.go:46]: agreed

/hold

Comment (gabemontero) [reply test/extended/builds/run_fs_verification.go:46]: do you want to at least approve @bparees so someone else can do final review and lgtm when we are ready ? 

or would you prefer to be the one that gates merging?

**Comment by gabemontero** [comment]
I've now added a test that mimics the chmod /run that was failing with the DPTP BC

This test should fail prior to https://github.com/openshift/builder/pull/206 merging

**Comment by gabemontero** [comment]
New test failed as expected with openshift/builder#206 missing 

```
2021-01-21T23:51:51.935093310Z F0121 23:51:51.934983       1 helpers.go:115] error: build error: error building at STEP "RUN chmod -R uga+rwx /run": error while running runtime: exit status 1
```

Existing test passed with new symlink stuff

/retest

**Comment by gabemontero** [comment]
failures in e2e-gcp, e2e-gcp-builds, e2e-gcp-fips all related to my new test for the dptp bc that should work again once https://github.com/openshift/builder/pull/206 merges

**Comment by gabemontero** [comment]
> failures in e2e-gcp, e2e-gcp-builds, e2e-gcp-fips all related to my new test for the dptp bc that should work again once [openshift/builder#206](https://github.com/openshift/builder/pull/206) merges

OK builder 206 has merged.  Hopefully I've waited long enough for the CI pipeline stuff to have promoted it through

/retest

reminder, the file system test should fail as is ... we can adjust the expected listing based on the reduced mounts

the test platform docker build approximation with the chmod on /run should now pass

**Comment by gabemontero** [comment]
OK failures / successes as expected

Pushing up changes now for all success

**Comment by gabemontero** [comment]
new fs listing test for fips needs a tweak

**Comment by gabemontero** [comment]
all build.sh tests passed in cmd

/skip

sig-scheduling and sig-network failures in serial

/retest

**Comment by gabemontero** [comment]
and build/fips clean

/assign @adambkaplan

**Comment by xiuwang** [comment]
/bugzilla cc-qa
/label qe-approved
Run the e2e suite passed with 4.7.0-0.nightly-2021-01-27-002938 cluster(included fix openshift/builder#206 )

**Comment by vrutkovs** [comment]
This breaks builds on OKD:
```
/run:
lock
rhsm
secrets

/run/lock:

/run/rhsm:

/run/secrets:
```
https://storage.googleapis.com/origin-ci-test/logs/release-openshift-okd-installer-e2e-aws-4.7/1355560957442527232/build-log.txt

**Pull request merged**